app/controllers/api/v1x0/icons_controller.rb

```diff
@@ -19,15 +19,26 @@ def destroy
 
       def update
         icon = Icon.find(params.require(:id))
+        if params.key?(:content)
+          params.require(:content)
+          params.require(:filename)
+          new_image = Image.new(:content   => params.delete(:content),
+                                :extension => params.delete(:filename).split(".").last)
+          image_id = Catalog::DuplicateImage.new(new_image).process.image_id
```

Comment:
Is there any case where `image_id` would be nil here?

Reply:
The only way that it would be nil would be if there was no match found AND the persist failed here: https://github.com/lindgrenj6/catalog-api/blob/602b55234d0a9cc38b37557ae66b1ad65cc2ac6f/app/services/catalog/duplicate_image.rb#L21
that would bubble up an exception at least so we know that something is wrong.